torch/fx/experimental/symbolic_shapes.py

```diff
@@ -5344,22 +5344,37 @@ def _maybe_evaluate_static(
 
         # axioms with compute hint NYE
         assert not compute_hint or not axioms
-
         expr = self.simplify(expr)
 
         if compute_hint:
             expr = expr.xreplace(self.var_to_val).xreplace(self.unbacked_var_to_val)
 
         expr = canonicalize_bool_expr(expr)
 
+        def try_simplify_floor_div(axioms) -> None:
+            new_items = {}
+            for k, v in axioms.items():
+                # A FloorDiv in implications could have became CleanDiv at this point, due to new facts
+                # to the shapeEnv. This handles such issue but its not ideal. This is the only expression
+                # simplification that depends on the global state of shape env.
+                # TODO try to get rid of CleanDiv since it breaks the invariant thats simplifications of sympy
+                # expressions only depend on the expression itself.
+                if k.has(FloorDiv):
+                    new_items.update({self.simplify(k): v})
+            axioms.update(new_items)
+
         # Pattern matching
         if axioms is None:
+            if expr.has(CleanDiv):
+                try_simplify_floor_div(self.axioms)
```

Comment:
Why don't you just unconditionally run this after the if statement

Reply:
I am not sure if its faster to  expr.has(CleanDiv): or iterate on the axioms.  for each expression
but if we do change were we only run that when a new divisibility constraint is added. then probably always running it is ok

Reply:
Actually removing expr.has(CleanDiv) makes it very slow!